### ISSUE TITLE
Couldn't create pulseaudio client: No valid directory for Pulse RuntimePath found

### ISSUE THREAD

**Issue by pinootto** [lawl/NoiseTorch#217]
Hello,

I am trying noisetorch 0.11.4 on Slackware 14.2, but I get this error:

```
$ noisetorch -log
2022/01/29 11:45:49 Application starting. Version: 0.11.4 (official)
2022/01/29 11:45:49 CAP_SYS_RESOURCE: true
2022/01/29 11:45:49 Checking if config needs to be initialized
2022/01/29 11:45:49 Couldn't resolve $XDG_CONFIG_HOME falling back to '/home/giovanni/.config'
2022/01/29 11:45:49 Wrote temp librnnoise to: /tmp/librnnoise-1774658256.so
2022/01/29 11:45:49 Couldn't resolve $XDG_CONFIG_HOME falling back to '/home/giovanni/.config'
Couldn't create pulseaudio client: No valid directory for Pulse RuntimePath found
2022/01/29 11:45:49 Deleted temp librnnoise: /tmp/librnnoise-1774658256.so
```

**Comment by lawl** [comment]
Can yoi take a look at [this](https://github.com/lawl/NoiseTorch/issues/212), i recently wrote some relatively in-depth suggestion on how to debug this there.

**Comment by pinootto** [comment]
Thanks. I have followed the information on that issue.
I have found the following:

```
giovanni@slackserver:~$ cd .config/pulse/
giovanni@slackserver:~/.config/pulse$ v
total 80
-rw-r--r-- 1 giovanni users 20480 Nov  2 14:00 7fde31b4ae42ff8115749936583a8f07-card-database.tdb
-rw-r--r-- 1 giovanni users     9 Jan 29 12:02 7fde31b4ae42ff8115749936583a8f07-default-sink
-rw-r--r-- 1 giovanni users     8 Jan 29 12:02 7fde31b4ae42ff8115749936583a8f07-default-source
-rw-r--r-- 1 giovanni users 24576 Sep 20 22:51 7fde31b4ae42ff8115749936583a8f07-device-manager.tdb
-rw-r--r-- 1 giovanni users 12288 Jan 27 14:48 7fde31b4ae42ff8115749936583a8f07-device-volumes.tdb
lrwxrwxrwx 1 giovanni users    23 Nov 27  2016 7fde31b4ae42ff8115749936583a8f07-runtime -> /tmp/pulse-PKdhtXMmr18n/
-rw-r--r-- 1 giovanni users 12288 Jan 28 22:12 7fde31b4ae42ff8115749936583a8f07-stream-volumes.tdb
-rw------- 1 giovanni users   256 Nov 27  2016 cookie
```
then I started the command setting the PULSE_RUNTIME_PATH:

`PULSE_RUNTIME_PATH=/home/giovanni/.config/pulse/7fde31b4ae42ff8115749936583a8f07-runtime strace noisetorch -l`

and I got a new error:

```
newfstatat(AT_FDCWD, "/home/giovanni/.config/noisetorch", {st_mode=S_IFDIR|0700, st_size=25, ...}, 0) = 0
newfstatat(AT_FDCWD, "/home/giovanni/.config/noisetorch/config.toml", {st_mode=S_IFREG|0644, st_size=166, ...}, 0) = 0
openat(AT_FDCWD, "/tmp/librnnoise-3238466821.so", O_RDWR|O_CREAT|O_EXCL|O_CLOEXEC, 0600) = 3
epoll_ctl(4, EPOLL_CTL_ADD, 3, {EPOLLIN|EPOLLOUT|EPOLLRDHUP|EPOLLET, {u32=85409416, u64=140518530432648}}) = -1 EPERM (Operation not permitted)
write(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\200\22\0\0\0\0\0\0"..., 143592) = 143592
openat(AT_FDCWD, "/home/giovanni/.config/noisetorch/config.toml", O_RDONLY|O_CLOEXEC) = 7
epoll_ctl(4, EPOLL_CTL_ADD, 7, {EPOLLIN|EPOLLOUT|EPOLLRDHUP|EPOLLET, {u32=85409416, u64=140518530432648}}) = -1 EPERM (Operation not permitted)
fstat(7, {st_mode=S_IFREG|0644, st_size=166, ...}) = 0
read(7, "Threshold = 95\nDisplayMonitorSou"..., 512) = 166
read(7, "", 346)                        = 0
close(7)                                = 0
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 7
connect(7, {sa_family=AF_LOCAL, sun_path="/home/giovanni/.config/pulse/7fde31b4ae42ff8115749936583a8f07-runtime/native"}, 79) = 0
epoll_ctl(4, EPOLL_CTL_ADD, 7, {EPOLLIN|EPOLLOUT|EPOLLRDHUP|EPOLLET, {u32=85409416, u64=140518530432648}}) = 0
getsockname(7, {sa_family=AF_LOCAL, NULL}, [2]) = 0
getpeername(7, {sa_family=AF_LOCAL, sun_path="/tmp/pulse-PKdhtXMmr18n/native"}, [33]) = 0
futex(0xc000100150, FUTEX_WAKE_PRIVATE, 1) = 1
newfstatat(AT_FDCWD, "", 0xc000123b08, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/home/giovanni/.config/pulse/cookie", {st_mode=S_IFREG|0600, st_size=256, ...}, 0) = 0
openat(AT_FDCWD, "/home/giovanni/.config/pulse/cookie", O_RDONLY|O_CLOEXEC) = 8
epoll_ctl(4, EPOLL_CTL_ADD, 8, {EPOLLIN|EPOLLOUT|EPOLLRDHUP|EPOLLET, {u32=85409184, u64=140518530432416}}) = -1 EPERM (Operation not permitted)
fstat(8, {st_mode=S_IFREG|0600, st_size=256, ...}) = 0
read(8, "v\256\306\351\32\265r\330\2<\376M0\0265\255\276i$\10`t\256(= \364r\254\340\203 "..., 512) = 256
read(8, "", 256)                        = 0
close(8)                                = 0
futex(0xa657f0, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
Couldn't create pulseaudio client: pulseAudio server supports version 31 but minimum required is 32
epoll_pwait(4, [], 128, 0, NULL, 0)     = 0
epoll_pwait(4,  <unfinished ...>
+++ exited with 1 +++
```

**Comment by lawl** [comment]
> Couldn't create pulseaudio client: pulseAudio server supports version 31 but minimum required is 32

It seems you're running a really old version of pulseaudio. Unfortunately i have no plans of adding support for old pulseaudio versions. The last person affected by this was running ubuntu 18.something and this literally only seems to affect a handful of people.

The only solution here is for you to upgrade pulseaudio.

**Comment by pinootto** [comment]
OK, thanks. I will try to upgrade pulseaudio.
My current version is:
```
$ pulseaudio --version
pulseaudio 9.0-rebootstrapped
```
What is the minimum required version to run noisetorch?

**Comment by lawl** [comment]
As the error message sais, whatever supports protocol version 32.
https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/blob/master/PROTOCOL

    ## v32, implemented by >= 10.0

**Comment by pinootto** [comment]
OK, thank you very much for your help.
